commons/src/main/java/org/infinispan/commons/util/InfinispanCollections.java

```diff
@@ -301,15 +301,28 @@ public int size() {
     * @see #EMPTY_LIST
     */
    @SuppressWarnings("unchecked")
-   public static final <T> List<T> emptyList() {
+   public static <T> List<T> emptyList() {
       return EMPTY_LIST;
    }
 
-   public static final <T> boolean containsAny(Collection<T> haystack, Collection<T> needle) {
+   public static <T> boolean containsAny(Collection<T> haystack, Collection<T> needle) {
       for (T element : needle) {
          if (haystack.contains(element))
             return true;
       }
       return false;
    }
+
+   public static <T> T findAnyCommon(Collection<T> c1, Collection<T> c2) {
```

Comment:
Didn't get what you mean. ```findAny()``` returns a random element and I want a random element common to both collections :)